### PR TITLE
GL shader fixes on macOS

### PR DESCRIPTION
**Pull request by garbear**
With these fixes (and the manual glsl folder copy), I can view a list of shaders in the Video Filters dialog. Shader presets fail to load, but I need to rebuild game.shader.presets with debug symbols to find out why. At least shader presets fail gracefully, and show a NN image when focused.

### CODE REVIEW

**Comment by gusandrianos** [comment]
Getting a white screen with your changes

File: xbmc/cores/RetroPlayer/shaders/gl/ShaderGL.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ bool CShaderGL::Create(const std::string& shaderSource, const std::string& shade
   else
     defineFragment = "#define FRAGMENT\n#define PARAMETER_UNIFORM\n";
 
-  if (m_shaderSource.rfind("#version", 0) == 0)
+  if (m_shaderSource.rfind("#version", 0) == std::string::npos)
```
Comment (gusandrianos) [inline xbmc/cores/RetroPlayer/shaders/gl/ShaderGL.cpp]: This change completely breaks functionality

**Review by gusandrianos — CHANGES_REQUESTED** [review]
Also, I deleted this from my branch, you should probably do the same.

File: xbmc/cores/RetroPlayer/shaders/gl/ShaderGL.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/cores/RetroPlayer/shaders/gl/ShaderGL.cpp]: I misread the code. This change is incorrect.

**Comment by garbear** [comment]
Rebased and removed the incorrect commit.

**Pull request merged**